product-service/handlers/addProduct.ts

```diff
@@ -0,0 +1,76 @@
+import {Context, APIGatewayEvent, APIGatewayProxyResult} from "aws-lambda";
+import {corsHeaders} from "../constants/headers";
+import {Client} from "pg";
+import {v4 as uuidv4} from 'uuid';
+
+const {PG_HOST, PG_PORT, PG_DATABASE, PG_USERNAME, PG_PASSWORD} = process.env;
+const dbOptions = {
+    host: PG_HOST,
+    port: PG_PORT,
+    database: PG_DATABASE,
+    user: PG_USERNAME,
+    password: PG_PASSWORD,
+    ssl: {rejectUnauthorized: false},
+    connectionTimeoutMillis: 5000
+};
+
+export async function addProduct(
+    event: APIGatewayEvent,
+    context: Context
+): Promise<APIGatewayProxyResult> {
+    console.log(event);
+    console.log(event.body);
+    const {
+        count,
+        description,
+        id,
+        price,
+        title,
+        image_url,
+        image_title
+    } = JSON.parse(event.body);
+
+    const client = new Client(dbOptions);
+    await client.connect();
+
+    try {
+        await client.query('BEGIN');
+
+        const product = await client.query(`
+            insert into products (id, title, description, price, image_url, image_title)
+            values ('${id || uuidv4()}',
```

Comment:
Good point. It is not obligatory for API users to pass an id for product creation, but they have an option to do so. Lambda will generate a UUID for you.
SQL script has uuid_generate_v4() in case direct insert to DB.